src/rust/src/asn1.rs

```diff
@@ -77,21 +78,38 @@ struct DssSignature<'a> {
     s: asn1::BigUint<'a>,
 }
 
-pub(crate) fn big_asn1_uint_to_py<'p>(
+pub(crate) fn big_byte_slice_to_py_int<'p>(
     py: pyo3::Python<'p>,
-    v: asn1::BigUint<'_>,
+    v: &'_ [u8],
 ) -> pyo3::PyResult<&'p pyo3::PyAny> {
+    warn_if_negative(py, v)?;
```

Comment:
I still object to this being part of the behavior of `big_byte_slice_to_py_int.

Functions behavior should be predictable and well factored -- generic functions like type conversion shouldn't have context specific warnings like this. Just put `warn_if_negative` calls in the serial accessors. And please also name it `warn_if_negative_serial` based on the warning message.